### ISSUE TITLE
Fix AD0001 on S2222: IndexOutOfRange for lambda in try/finally

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#5916]
When analyzing inner CFG for lambda, try/finally handler region is picked up from outer CFG. Found via dotnet ITs Nancy.

`LocksReleasedAllPaths.Monitor.TryCatch.cs`
```cs
        public void TryFinaly_InLoop(bool condition)
        {
            Monitor.Enter(obj);
            try
            {
                Action a = () =>
                {
                    M();
                    Monitor.Enter(obj);
                    Monitor.Exit(obj);
                };
                a();
            }
            finally
            {
                M();
            }

            if (condition)
            {
                Monitor.Exit(obj);
            }

            void M() { }
        }
```

```
Message: 
Expected diagnostics {LocksReleasedAllPaths.Monitor.Conditions.cs(19,13): warning S2222: Unlock this lock along all executions paths of this method., LocksReleasedAllPaths.Monitor.Conditions.cs(28,13): warning S2222: Unlock this lock along all executions paths of this method., …81 more…} to not have any items matching (d.Id == "AD0001"), but found {error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.SymbolicExecutionRunner' threw an exception of type 'SonarAnalyzer.SymbolicExecution.SymbolicExecutionException' with message 'Error processing method:  ## Method file: LocksReleasedAllPaths.Monitor.TryCatch.cs ## Method line: 358,16 ## Inner exception: System.IndexOutOfRangeException: Index was outside the bounds of the array. ##    at System.Collections.Immutable.ImmutableArray`1.get_Item(Int32 index) ##    at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.ExceptionSuccessors(ExplodedNode node, ExceptionState exception, ControlFlowRegion tryRegion)+MoveNext() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 222 ##    at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.ProcessOperation(ExplodedNode node)+MoveNext() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 204 ##    at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.Execute() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 88 ##    at SonarAnalyzer.Rules.SymbolicExecutionRunnerBase.AnalyzeRoslyn(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext, Boolean isTestProject, Boolean isScannerRun, SyntaxNode body, ISymbol symbol) in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\Rules\SymbolicExecutionRunnerBase.cs:line 94'.
Exception occurred with following context:
Compilation: project0
SyntaxTree: LocksReleasedAllPaths.Monitor.TryCatch.cs
SyntaxNode: () => { M(); Monitor.Enter(obj) ... [ParenthesizedLambdaExpressionSyntax]@[8548..8695) (357,27)-(362,17)

SonarAnalyzer.SymbolicExecution.SymbolicExecutionException: Error processing method:  ## Method file: LocksReleasedAllPaths.Monitor.TryCatch.cs ## Method line: 358,16 ## Inner exception: System.IndexOutOfRangeException: Index was outside the bounds of the array. ##    at System.Collections.Immutable.ImmutableArray`1.get_Item(Int32 index) ##    at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.ExceptionSuccessors(ExplodedNode node, ExceptionState exception, ControlFlowRegion tryRegion)+MoveNext() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 222 ##    at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.ProcessOperation(ExplodedNode node)+MoveNext() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 204 ##    at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.Execute() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 88 ##    at SonarAnalyzer.Rules.SymbolicExecutionRunnerBase.AnalyzeRoslyn(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext, Boolean isTestProject, Boolean isScannerRun, SyntaxNode body, ISymbol symbol) in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\Rules\SymbolicExecutionRunnerBase.cs:line 94
 ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at System.Collections.Immutable.ImmutableArray`1.get_Item(Int32 index)
   at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.ExceptionSuccessors(ExplodedNode node, ExceptionState exception, ControlFlowRegion tryRegion)+MoveNext() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 222
   at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.ProcessOperation(ExplodedNode node)+MoveNext() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 204
   at SonarAnalyzer.SymbolicExecution.Roslyn.RoslynSymbolicExecution.Execute() in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\SymbolicExecution\Roslyn\RoslynSymbolicExecution.cs:line 88
   at SonarAnalyzer.Rules.SymbolicExecutionRunnerBase.AnalyzeRoslyn(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext, Boolean isTestProject, Boolean isScannerRun, SyntaxNode body, ISymbol symbol) in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\Rules\SymbolicExecutionRunnerBase.cs:line 94
   --- End of inner exception stack trace ---
   at SonarAnalyzer.Rules.SymbolicExecutionRunnerBase.AnalyzeRoslyn(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext, Boolean isTestProject, Boolean isScannerRun, SyntaxNode body, ISymbol symbol) in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\Rules\SymbolicExecutionRunnerBase.cs:line 99
   at SonarAnalyzer.Rules.SymbolicExecutionRunnerBase.Analyze(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext, SyntaxNode body, ISymbol symbol) in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\Rules\SymbolicExecutionRunnerBase.cs:line 74
   at SonarAnalyzer.Rules.CSharp.SymbolicExecutionRunner.<>c__DisplayClass8_0.<Initialize>b__4(SyntaxNodeAnalysisContext c) in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.CSharp\Rules\SymbolicExecution\SymbolicExecutionRunner.cs:line 102
   at SonarAnalyzer.Helpers.DiagnosticAnalyzerContextHelper.<>c__DisplayClass1_0`1.<RegisterSyntaxNodeActionInNonGenerated>b__0(SyntaxNodeAnalysisContext c) in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\Helpers\DiagnosticAnalyzerContextHelper.cs:line 42
   at SonarAnalyzer.Helpers.SonarAnalysisContext.<>c__DisplayClass47_0`1.<RegisterContextAction>b__0(TContext c) in C:\Projects\sonar-dotnet\analyzers\src\SonarAnalyzer.Common\Helpers\SonarAnalysisContext.cs:line 243
   at Microsoft.Cod
... <truncated>
```
